tests/wp_viewporter.cpp

```diff
@@ -104,21 +108,21 @@ class WpViewporterTest : public wlcs::InProcessServer
         }
 
         // Moving any further should take us out of the surface
-        motion_received = false;
+        *motion_received = false;
 
         client.add_pointer_leave_notification(
-            [&](auto)
+            [pointer_entered](auto)
             {
-                pointer_entered = false;
-                return false;
-            });
-        client.add_pointer_motion_notification(
-            [&](auto, auto)
-            {
-                motion_received = true;
+                *pointer_entered = false;
                 return false;
             });
 
+        client.add_pointer_motion_notification([motion_received](auto, auto)
+        {
+            *motion_received = true;
+            return false;
+        });
+
         pointer.move_by(1, 1);
         client.dispatch_until([&]() { return !pointer_entered || motion_received; });
```

Comment:
The trouble with pointers is that they implicitly convert to bool.
```suggestion
        client.dispatch_until([&]() { return !*pointer_entered || *motion_received; });
```